gui/src/components/OnboardingWizard/OnboardingCarousel.tsx

```diff
@@ -0,0 +1,85 @@
+import KeyboardArrowLeft from "@mui/icons-material/KeyboardArrowLeft";
+import KeyboardArrowRight from "@mui/icons-material/KeyboardArrowRight";
+import { Container, IconButton, Stack } from "@mui/material";
+import Box from "@mui/material/Box";
+import MobileStepper from "@mui/material/MobileStepper";
+import Typography from "@mui/material/Typography";
+import { useTheme } from "@mui/material/styles";
+
+import { Step } from "./Steps";
+
+interface Props {
+  steps: Step[];
+  activeStep: number;
+  handleClose: () => void;
+  handleNext: () => void;
+  handleBack: () => void;
```

Comment:
I believe `unknown` is the recommended approach, instead of `void`. Right?

Reply:
But that's the point; you don't care if this function returns something, so why force it to return void? The beauty of `unknown` is that even if the function returns something, you can't do anything with the return value.
For instance, the parent component may also use the `handleNext` for other components where it has to return something. By setting the return type to void, you don't allow for that, even if it doesn't matter for your component.

Reply:
semantically, `void` makes more sense here. `unknown` doesn't mean you can't do anything with it. you can still assign the result to a binding or put it in an object thinking that it's not null/undefined

in practice the only problem is it forces you to not return anything, meaning that `() => foo()` has to actually be `() => { foo(); }`, which is more cumbersome

Reply:
What do you mean? I don't understand what you said. [You cannot use the return value of a function that returns 'unknown'](https://www.typescriptlang.org/docs/handbook/2/functions.html#unknown). That's the goal, really; you don't care what the function returns, no matter if it returns something or nothing at all, you don't care. When you use void, you are explicitly saying that you care. In my opinion, in this scenario, you shouldn't care.

Reply:
That's not true. we're getting really into semantics here, but: unknown is exactly what the name suggests. it's unknown. you can't call method functions on it because you don't know which ones exist. but you can still use the value:

```
function u(): unknown {
    return 1
}
// this works
const x = u();
console.log({ foo: x});
```

the semantics of `void` are precisely to prevent you from even having `return 1` in the first place
saying "I explicitly don't return anything" is very different from saying "I may return something, but refuse to tell you what it is"

Reply:
Interesting, I'm not sure why I thought you couldn't use an unknown, but it makes sense:

```ts
function one() {
    return "world"
}

function two(fn: () => unknown) {
    const value = fn();

    if (typeof value === 'string') {
        console.log("It's a string!")
    }
}

two(one)
```

Do you know what the funny part is? You can do the same with a void:

```
function one() {
    return "world"
}

function two(fn: () => void) {
    const value = fn();

    if (typeof value === 'string') {
        console.log("It's a string!")
    }
}

two(one)
```

So now I no longer know what to think 😅

Reply:
hmm, didn't know that weird trick worked. my experience with `void` is only that TS prevents you from having a `return` statement inside the function (or an implicit return with arrow functions)

but apparently you can ask eslint to catch that, at least:
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-confusing-void-expression.md

Reply:
Nice, we should enable that. For the sake of this change, if having here void doesn't require the `() => { foo(); }` on the parent, I think it's ok, if not, I still think it's probably better to use the unknown.

Reply:
@naps62  I will take this one . Gimme the access

Reply:
@naps62  I will take this one . Gimme the access